app/process/watcher/evm/watcher.go

```diff
@@ -85,58 +91,95 @@ func NewWatcher(
 		log.Tracef("[%s] - Updated Transfer Watcher timestamp to [%s]", contracts.Address(), timestamp.ToHumanReadable(startBlock))
 	}
 	return &Watcher{
-		repository:  repository,
-		contracts:   contracts,
-		evmClient:   evmClient,
-		logger:      c.GetLoggerFor(fmt.Sprintf("EVM Router Watcher [%s]", contracts.Address())),
-		mappings:    mappings,
-		targetBlock: targetBlock,
-		validator:   validator,
+		repository:    repository,
+		contracts:     contracts,
+		evmClient:     evmClient,
+		logger:        c.GetLoggerFor(fmt.Sprintf("EVM Router Watcher [%s]", contracts.Address())),
+		mappings:      mappings,
+		targetBlock:   targetBlock,
+		validator:     validator,
+		sleepDuration: sleepDuration,
 	}
 }
 
 func (ew *Watcher) Watch(queue qi.Queue) {
-	go ew.listenForEvents(queue)
+	go ew.beginWatching(queue)
 
-	ew.processPastLogs(queue)
 	ew.logger.Infof("Listening for events at contract [%s]", ew.contracts.Address())
 }
 
-func (ew Watcher) processPastLogs(queue qi.Queue) {
+func (ew Watcher) beginWatching(queue qi.Queue) {
 	fromBlock, err := ew.repository.Get(ew.contracts.Address().String())
 	if err != nil {
 		ew.logger.Errorf("Failed to retrieve EVM Watcher Status fromBlock. Error: [%s]", err)
-		return
 	}
 
 	ew.logger.Infof("Processing events from [%d]", fromBlock)
 
+	for {
+		fromBlock, err := ew.repository.Get(ew.contracts.Address().String())
+		if err != nil {
+			ew.logger.Errorf("Failed to retrieve EVM Watcher Status fromBlock. Error: [%s]", err)
+			continue
+		}
+
+		currentBlock, err := ew.evmClient.BlockNumber(context.Background())
+		if err != nil {
+			ew.logger.Errorf("Failed to retrieve latest block number. Error [%s]", err)
+			time.Sleep(ew.sleepDuration)
+			continue
+		}
+
+		toBlock := int64(currentBlock - ew.evmClient.BlockConfirmations())
+		if fromBlock > toBlock {
+			time.Sleep(ew.sleepDuration)
+			continue
+		}
+
+		err = ew.processLogs(fromBlock, toBlock, queue)
+		if err != nil {
+			ew.logger.Errorf("Failed to process logs. Error: [%s].", err)
+			time.Sleep(ew.sleepDuration)
+			continue
```

Comment:
nit: we do not need those 2 lines of code

Reply:
I've left them on purpose, so that if the error is timeout related, it will pause for a bit, rather than trying again immediately

Reply:
Aren't we going to pause for a bit even if we remove those 2 lines because of line 197?